src/main/java/org/elasticsearch/cluster/action/index/NodeIndexDeletedAction.java

```diff
@@ -64,32 +75,56 @@ public void remove(Listener listener) {
     }
 
     public void nodeIndexDeleted(final ClusterState clusterState, final String index, final String nodeId) throws ElasticsearchException {
-        DiscoveryNodes nodes = clusterState.nodes();
+        final DiscoveryNodes nodes = clusterState.nodes();
         if (nodes.localNodeMaster()) {
-            threadPool.generic().execute(new Runnable() {
+            threadPool.generic().execute(new AbstractRunnable() {
+
+                @Override
+                public void onFailure(Throwable t) {
+                    logger.warn("failed to ack index store deleted for  index {}", t, index);
+                }
+
                 @Override
-                public void run() {
+                protected void doRun() throws Exception {
                     innerNodeIndexDeleted(index, nodeId);
+                    lockIndexAndAck(index, nodes, nodeId, clusterState);
+
                 }
             });
         } else {
             transportService.sendRequest(clusterState.nodes().masterNode(),
                     INDEX_DELETED_ACTION_NAME, new NodeIndexDeletedMessage(index, nodeId), EmptyTransportResponseHandler.INSTANCE_SAME);
+            threadPool.generic().execute(new AbstractRunnable() {
```

Comment:
hey guys I did some work a while ago on the delete index api acknowledgement in #4218 which is the only api left that doesn't use the "standard" acknowledgement code. That said we decided at that time not to migrate it to keep two different actions: one for deletion from cluster state, one for deletion from store. That said we try and wait for both. Not sure I follow these changes when it comes to how they affect acknowledgements, but If we want to make the two a single action,  can't we just use the standard acknowledgement code and drop the custom actions?